### DISCUSSION TITLE
Take node scheduler offline gracefully

### DISCUSSION THREAD

**Discussion by smbecker** [quartznet/quartznet#481] [General] [answered]
I have a cluster of job nodes running and my goal is to be able to take one of these nodes out of the cluster so I can restart it gracefully. I could stop the scheduler but this would kill the threadpool and any jobs currently executing, which I want to let those complete before shutting down. I could veto the jobs if they attempt to run on a node that is scheduled for shutdown but won't that skip the current job interval and reschedule the jobs for the next interval? This is also not desired. I want the node to finish executing any current jobs but prevent it from picking up any new jobs. Is this possible? 

If this isn't something already provided out of the box, I suppose that I could customize the threadpool to prevent it from showing any threads as available if the node is scheduled for shutdown. Would this work? Are there things that I would need to consider if I had to go this way?

**Comment by lahma** [comment, marked as answer]
This should be covered by Shutdown(bool waitForJobsToComplete), so just passing true should wait for jobs to complete and no new jobs are taken to processing.
